### ISSUE TITLE
🐛 BUG: Client side script resolution does not use Snowpack

### ISSUE THREAD

**Issue by WDeenik** [snowpackjs/astro#574]
### What package manager are you using?

yarn

### What operating system are you using?

Mac

### Describe the Bug

In my test project I am using a Snowpack alias to my components folder. Everything works fine that way, until I hydrate one of those components (a Svelte component) using `:load`. As soon as I do that, I get the error `src/pages/index.astro: could not find "MyComponent.svelte.js`. This error is thrown at https://github.com/snowpackjs/astro/blob/main/packages/astro/src/build.ts#L112.

With some help on Discord I figured out that everything works again if I do not use the Snowpack alias for that component. @natemoo-re on Discord: 
> Ooh that’s definitely a bug. We aren’t running the client side script resolution through Snowpack, but we should be.

### Steps to Reproduce

1. `npm init astro` using template Starter Kit
2. Add a Snowpack alias to the components folder
3. Add a component (Svelte in my case but I don't think it matters that much)
4. Add a page which hydrates that component on load. Import the component using the Snowpack alias.
5. You'll get an error that it cannot find your component during build.

Maybe even easier to pick a template with existing client-side components and just change the import to an alias.

**Comment by natemoo-re** [comment]
Thanks for filing this, I'll take a look!

**Comment by natemoo-re** [comment]
Update: This one is a bit trickier than I anticipated due to _when_ Astro extracts hydrated components, but it's still on my list!

**Comment by natemoo-re** [comment]
I've closed a number of issues as duplicates of this one. These will all be fixed in the new compiler because we are using Vite's `moduleGraph` to resolve final component URLs at runtime.

Aliases will be supported as will bare import specifiers.

**Comment by natemoo-re** [comment]
Just wanted to give an update! I can confirm that this is working in `astro@next`, which uses Vite under the hood!

Putting this in our v0.21 milestone to close when that release becomes stable.

**Comment by WDeenik** [comment]
Great, thank you! 🎉

**Comment by natemoo-re** [comment]
Following up on my previous comment, the team has decided to close out issues that have been confirmed as fixed by [astro@0.21.0-next.0](https://github.com/snowpackjs/astro/releases/tag/astro%400.21.0-next.0), [astro@0.21.0-next.1](https://github.com/snowpackjs/astro/releases/tag/astro%400.21.0-next.1), or [astro@0.21.0-next.2](https://github.com/snowpackjs/astro/releases/tag/astro%400.21.0-next.2). Our hope is that this will help the [v0.21 milestone](https://github.com/snowpackjs/astro/milestone/1) remain as actionable as possible.

To verify that this issue has been fixed, you may
- Open a new [`astro@next` project on StackBlitz](https://astro.new/minimal@next)
- Locally upgrade to the `astro@next` preview
    ```shell
    npm i astro@next
    # If using Framework renderers...
    npm i @astrojs/renderer-name@next
    ```

If you think we made a mistake, **please reopen this issue**!

If you run into any other problems with `astro@next`, please [open a new issue](https://github.com/snowpackjs/astro/issues/new?assignees=&labels=0.21&template=---01-bug-report.yml&title=%F0%9F%90%9B+BUG%3Aastro@next).

**Comment by flowforfrank** [comment]
For anyone stumbling upon this issue, if you still experience the problem even after disabling aliases, make sure you have the extension in the import. It failed to find it for me without it. (Even though the component rendered just fine)
```javascript
// 🔴 Don't
import Component from './components/header'

// 🟢 Do
import Component from './components/header.svelte'
```
